src/transaction/transaction.rs

```diff
@@ -104,8 +104,16 @@ impl Transaction {
             .await
     }
 
-    /// Create a `get_for_udpate` request.
-    /// Once resolved this request will pessimistically lock and fetch the value associated with the given key.
+    /// Create a `get for udpate` request.
+    /// Once resolved this request will pessimistically lock and fetch the value associated with the given key **at current timestamp**.
+    ///
+    /// Note: The behavior of this command does not follow snapshot isolation. It is similar to `select for update` in TiDB,
+    /// which is similar to that in MySQL. It reads the latest value (using current timestamp),
+    /// and the value is not cached in the local buffer.
+    /// So normal `get`-like commands after `get_for_update` will not be influenced, they still read values at `start_ts`.
+    ///
+    /// Different from `get`, this request does not distinguish between empty values and non-existent keys
```

Comment:
This is the behavior of TiDB. If we cache results of `get_for_update`
1. We cannot distinguish between empty values and non-existent keys. 
2. It's not snapshot isolation. Consider a normal `get`, `batch_get` or `scan` after a `get_for_update`, it will cause anomalies, since the results of the same request may be from different timestamps.

So in my opinion, `get_for_update` just provides a special request to (1) lock the key and (2) fetch the latest value, but should not affect other transactional requests.